core/controlplane/config/templates/stack-template.json

```diff
@@ -715,6 +737,27 @@
       "Type": "AWS::AutoScaling::LaunchConfiguration"
     },
     {{end}}
+    {{if .Experimental.NodeDrainer.Enabled }}
+    "{{.Controller.LogicalName}}NodeDrainerLHT" : {
+      "Properties" : {
+        "DisplayName" : "SNS topic for the {{.Controller.LogicalName}} node drainer ASG lifecycle hook"
+      },
+      "Type" : "AWS::SNS::Topic"
+    },
+    "{{.Controller.LogicalName}}NodeDrainerLH" : {
+      "Properties" : {
+        "AutoScalingGroupName" : {
+          "Ref": "{{.Controller.LogicalName}}"
+        },
+        "DefaultResult" : "CONTINUE",
+        "HeartbeatTimeout" : "{{.Experimental.NodeDrainer.DrainTimeoutInSeconds}}",
+        "LifecycleTransition" : "autoscaling:EC2_INSTANCE_TERMINATING",
+        "NotificationTargetARN" : { "Ref": "{{.Controller.LogicalName}}NodeDrainerLHT" },
+        "RoleARN" : { "Fn::GetAtt": [ "IAMRoleController", "Arn" ] }
```

Comment:
This topic is dedicated to that node draining hook, so I don't think allowing nodes themselves to access it would cause problems. Plus, it makes the stack JSON a bit smaller.

However, if you are strongly opposed to this, I can change it.